devito/types/sparse.py

```diff
@@ -113,92 +113,81 @@ def inject(self, *args, **kwargs):
         """
         return self.interpolator.inject(*args, **kwargs)
 
+    @cached_property
+    def _point_support(self):
+        return np.array(tuple(product(range(-self._radius + 1, self._radius + 1),
+                                      repeat=self.grid.dim)))
+
     @property
     def _support(self):
         """
         The grid points surrounding each sparse point within the radius of self's
         injection/interpolation operators.
         """
-        ret = []
-        for i in self.gridpoints:
-            support = [range(max(0, j - self._radius + 1), min(M, j + self._radius + 1))
-                       for j, M in zip(i, self.grid.shape)]
-            ret.append(tuple(product(*support)))
-        return tuple(ret)
+        max_shape = np.array(self.grid.shape).reshape(1, self.grid.dim)
+        minmax = lambda arr: np.minimum(max_shape, np.maximum(0, arr))
```

Comment:
maybe change this name?

Reply:
why?

Reply:
minmax is a bit ambiguous...?

Reply:
it's the min of the maximum -- pretty clear to me, resolving